climada/hazard/relative_cropyield.py

```diff
@@ -690,44 +753,74 @@ def calc_his_haz(his_file, file_props, input_dir=INPUT_DIR, bbox=BBOX, yearrange
             yearrange_mean (int tuple): year range for the historical mean
                 default: 1976 - 2005
 
-
         Return:
             haz_his (RelativeCropyield): historical hazard
             filename (string): name to save historical hazard
             hist_mean (array): historical mean of the historical hazard
-
     """
 
-
     haz_his = RelativeCropyield()
-    haz_his.set_from_single_run(input_dir=input_dir, filename=his_file, bbox=bbox,
-                                scenario=(file_props[his_file])['scenario'],
-                                yearrange=np.array([(file_props[his_file])['startyear'],
-                                                    (file_props[his_file])['endyear']]))
-
+    haz_his.set_from_isimip_netcdf(input_dir=input_dir, filename=his_file, bbox=bbox,
+                                scenario=file_props[his_file]['scenario'],
+                                yearrange=np.array([file_props[his_file]['startyear'],
+                                                    file_props[his_file]['endyear']]))
+    crop = file_props[his_file]['crop']
+    # combine subcrops if crop and combi_crop not the same:
+    if crop != file_props[his_file]['combi_crop']:
+        if crop in ('ri2', 'wwh'):
+            raise ValueError('Invalid subcrop type before combination')
+        elif crop == 'swh': # whe = swh+wwh
+            his_file2 = his_file.replace('_yield-swh-', '_yield-wwh-')
+        elif crop == 'ri1': # ric = ri1+ri2
+            his_file2 = his_file.replace('_yield-ri1-', '_yield-ri2-')
+        else:
+            his_file2 = None
+        if his_file2 and isfile(join(input_dir, his_file2)):
+            haz_his2 = RelativeCropyield()
+            haz_his2.set_from_isimip_netcdf(input_dir=input_dir, filename=his_file2, bbox=bbox,
+                                scenario=file_props[his_file]['scenario'],
+                                yearrange=np.array([file_props[his_file]['startyear'],
+                                                    file_props[his_file]['endyear']]))
+            if crop == 'swh':
+            # mask of winter wheat in spring wheat and vice versa:
```

Comment:
I am not sure whether I understand what the mask does. What does the file winter_and_spring_wheat_areas_phase3.nc4 contain?

Reply:
The masks in the NetCDF file divides all wheat growing areas globally in two distinct categories, either growing winter wheat (wwh) or spring wheat (swh). Since the hazard sets for wwh and swh are calculated for the whole globel ("all crops everywhere") we need to decide for each grid cell which one to take when combining wwh and swh into one combined crop wheat (whe). 

I will add this explanation as a comment in the script.